src/common/models/backend/safe_apps.rs

```diff
@@ -18,3 +19,24 @@ pub struct SafeAppProvider {
     pub url: String,
     pub name: String,
 }
+
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+#[serde(tag = "type")]
+pub enum SafeAppAccessControlPolicies {
+    #[serde(rename(deserialize = "NO_RESTRICTIONS"))]
+    NoRestrictions(SafeAppNoRestrictionsPolicy),
+    #[serde(rename(deserialize = "DOMAIN_ALLOWLIST"))]
```

Comment:
Did you try using `#[serde(rename_all = "SCREAMING_SNAKE_CASE")]` at the `enum` level? Or you want these `type`s have different value when you serialise them?

Reply:
> Or you want these types have different value when you serialise them?

If I understood you right, this is exactly what I want. I followed the pattern in https://github.com/gnosis/safe-client-gateway/blob/a36739ba10752ab34ded695a0de87df53c86495a/src/common/models/backend/transfers.rs#L8

Reply:
Ok so you want camelCased on serializaed and you expect them to be upper case when you deserialise them

Reply:
no, I expect them to be uppercase (or screaming snake case) when serialized too

Reply:
In that case you would need to implement according to my suggestion: 
```rust
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum SafeAppAccessControlPolicies {
   NoRestrictions(SafeAppNoRestrictionsPolicy),
   DomainAllowList(SafeAppDomainAllowlistPolicy)
}
```

Reply:
Just one thing, there are cleaner examples of polymorphism in the CGW, the one you used (`Transfers`) is quite old code and verbose. 

A cleaner newer example you can find here: https://github.com/gnosis/safe-client-gateway/blob/8cbaca58b181e63dcba1a22fb8c6b1d54c3a47db/src/common/models/backend/chains.rs#L45

After reviewing the original ticket in the config service, it seems like you don't even need `SafeAppNoRestrictionsPolicy` defined.